### ISSUE TITLE
Problem converting "workflow_by_code.py" to LSTM

### ISSUE THREAD

**Issue by bricard1** [microsoft/qlib#139]
I am just trying to run a bare minimum LSTM from the example given in 'workflow_by_code.py'.

As I understand it, there are 7 input variables for every time step, ("change", "open", "close", "volume", "factor", "high" and "low").

So why is d_feat not default =7? Furthermore, when I try to run workflow_by_code.py, and replace the model definition with:

 task = {
        "model": {
            "class": "LSTM",
            "module_path": "qlib.contrib.model.pytorch_lstm",
            "kwargs": {
            "GPU": 0,
            "d_feat": 2,
            "metric": 'loss',
            "n_epoch":5                                                                                                             }

Or any different combination of d_feat, the model runs but I only get error values in terms of 'nan'. I'm not sure why it still runs but returns values of 'nan' (I tried with d_feat = 1,2,6,158... i'm not sure what the 'proper value should be if not 7).

Output: 
[1740937:MainThread](2020-12-20 14:38:07,235) INFO - qlib.LSTM - [pytorch_lstm.py:240] - training...
[1740937:MainThread](2020-12-20 14:38:07,235) INFO - qlib.LSTM - [pytorch_lstm.py:244] - Epoch0:
[1740937:MainThread](2020-12-20 14:38:07,235) INFO - qlib.LSTM - [pytorch_lstm.py:245] - training...
[1740937:MainThread](2020-12-20 14:38:11,819) INFO - qlib.LSTM - [pytorch_lstm.py:247] - evaluating...
[1740937:MainThread](2020-12-20 14:38:13,947) INFO - qlib.LSTM - [pytorch_lstm.py:250] - train nan, valid nan

Any help would be appreciated. I feel like there is just something small I am missing. Do I still use DatasetH?

**Comment by bricard1** [comment]
Nevermind: If anybody has the same problem, just check the parameters in the benchmark workflows and copy those.

**Comment by bricard1** [comment]
Actually, I thought it would work, but I still get Nans, even when running 
'qrun benchmarks/LSTM/workflow_confid_lstm_Alpha158.yml'

even though the qrun for LightGBM still works.

**Comment by Derek-Wds** [comment]
Hi @bricard1, thanks for the issue. 

Can you provide more information about your environments of running the LSTM model by `qrun benchmarks/LSTM/workflow_confid_lstm_Alpha158.yml`? I cannot reproduce your situation since LSTM workflow works fine on my side.

![Screenshot from 2020-12-21 10-13-23](https://user-images.githubusercontent.com/26081991/102732281-3c7d2080-4375-11eb-9a09-345500ed42c3.png)

My environments are: Ubuntu 20.04, Python 3.8.3 (Anaconda), qlib 0.6.0.dev.

> Also, I do not understand the output loss, which are given as negative numbers when running GRU or LSTM models, but positive when running LightGBM.

Sorry for the confusion of the output of LSTM and GRU model. The outputs numbers are actually not losses, but are the negation of losses. You can refer to the line [here](https://github.com/microsoft/qlib/blob/main/qlib/contrib/model/pytorch_lstm.py#L153). Larger numbers you see indicate lower loss of the models. Apparently, the numbers are always negative as long as the losses of the model are positive.

We will try to update the logged information to make it for clear. Thanks for pointing it out!

**Comment by bricard1** [comment]
hi @Derek-Wds , I think I figured out the problem. Even though I installed under 3.7, qrun run under 3.6 because of a dependency error with package 'imp'. I think if I just upgrade to 3.8 the problem will be fixed.

**Comment by you-n-g** [comment]
@Derek-Wds  I think it will better if we can add the env info that has been tested  for  `run_all_models.py`
